### DISCUSSION TITLE
[Feature] Filter on pictures taken on a specific date in all years

### DISCUSSION THREAD

**Discussion by artus42** [immich-app/immich#10684] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be great to filter on all pictures which were taken at a specific day and month of any year. 
E.g. To display all the anniverseries, birthdays, memorial days, etc.
It almost exactly what is displayed on the start page for the current date but selectable from a user for a specific day and month.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by samluxh** [comment]
Hi there,

seems to be the same feature request, so I am not creating a new one.

I’m currently using immich and really enjoying it! However, I’ve noticed that it still seems impossible to filter by just a month (for example, August across all years) or by a year without specifying a particular day or date.

Is this functionality still not available, or is there a way to achieve this that I might have missed? On iOS you can do this in the native Gallery app.

Thanks in advance for your help, and thanks for all the great work on immich!

**Comment by avanrider** [comment]
Adding support for this a year later. (I searched but didn't find this). 
I'd really like it implemented, and even a shortcut to it for "On This Day" added.

**Comment by neosy** [comment]
Hi there, 
I use the app with pleasure. It’s a pity that this feature doesn’t exist yet — I really miss it.

**Comment by valerauko** [comment]
I'd love to see this feature too.

It'd be nice to have a way to find photos from certain repeating dates.

For example: 
* in March (regardless of year)
* 27th June (regardless of year)
* a 29th (regardless of year or month)

I'd love to use this for example when looking for birthday photos, or when compiling some calendar from photos from each month in the past.

I tried using the machine learning feature and search like "taken in january", but sadly it isn't exact enough (for example returns snowy photos taken in other months because I guess it assumes)

**Comment by PercentageCorporation** [comment]
I would like this feature also, but I would also like a way to build custom queries and have the pictures that meet those criteria displayed.

**Comment by moamarch** [comment]
I would like to extend this idea a bit further.

First, the **Memories** feature only shows memories for the current day. If a memory is missed, there is currently no easy way to revisit it later.

For comparison, OneDrive allows users to browse memories from previous days, making it easier to catch up on missed memories.

Additionally: 
Instead of only supporting day + month without year, Immich could support partial date filtering and recurring date patterns in general.

## Recurring Date Filters

It would also be useful to support recurring calendar patterns.

Examples:

- Day of week selection (Monday, Tuesday, etc.)
- Weekend selection
- Multiple weekdays (e.g. Friday and Saturday)
- Combinations with date ranges

### Examples

| Filter | Result |
|----------|----------|
| Friday | All photos taken on Fridays, because that is when our club meets regularly. |
| Weekend | All photos taken on Saturdays and Sundays |
| Friday during summer | All photos taken on Fridays in summer months |

## Benefits
With partial date filtering, users could effectively generate their own memories instead of being limited to the current day's automatically generated memory.

- Makes missed memories accessible again.
- Enables powerful photo discovery workflows.
- Helps users find recurring events and traditions.
- Makes it easier to explore large photo libraries.
- Requires only metadata that already exists in the database.
- Complements the existing Memories feature without changing its current behavior.
